External Code/SCPinions/SCPacketUtility/SCTracerouteUtility.m

```diff
@@ -85,6 +84,11 @@ -(void)stop{
     self.packetUtility = nil;
 }
 
+-(void)forcedTimeout {
+    if ( (self.delegate != nil) && [self.delegate respondsToSelector:@selector(tracerouteDidTimeout:)]) {
```

Comment:
@apike Good point. I've done it forever, there is reason behind it but its arguable:

https://stackoverflow.com/questions/6479267/objective-c-why-check-nil-before-respondstoselector